src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -687,19 +693,47 @@ private void WriteXmpChunk(Stream stream, ImageMetadata meta)
                 PngConstants.XmpKeyword.CopyTo(payload);
                 int bytesWritten = PngConstants.XmpKeyword.Length;
 
-                // Write the iTxt header (all zeros in this case)
+                // Write the iTxt header (all zeros in this case).
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
```

Comment:
Not directly related to this PR, but
```suggestion
                Span<byte> iTxtHeader = payload.Slice(bytesWritten);
                iTxtHeader[4] = 0;
                iTxtHeader[3] = 0;
                iTxtHeader[2] = 0;
                iTxtHeader[1] = 0;
                iTxtHeader[0] = 0;
                bytesWritten += 5;
```
will avoid extra bound checks for each element access, except for the first one (index 4). Cf. [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBuIGYACMxgYUYG8bGfmniUjALLkAFAGUADtgB2AHmABPDDAB8jaYoA2EbABM0jAJYyMjJStwB1KEYwqZASkbdeXar08bs23XoDaFjDWtvYwMgDUEQC6jAC8jAAMANyuXt6++oHKwTZ2DlGxCSlpXpo6WUEh+eGF8UmpHukZFQFVeWGRMfUlTenlftmWHQXdxY2eAL6lpQzMgkKkEtLyQeoD+oYmZu2hDs6l7s2MUrIKOepGACoIGAASMPowUPUbegB04lpGYDCiuzUnBNmtdbg8nlB/Cgig1Sp5QfdHnpnv56DDescEeDkZDSOjgeksUiUeR8XDeESIf5EmS+oxpnTZvwFvRlmc1i0/FtTOYctVOgc6UdmqdVhdjDdERDXj5Wp9vr9/nyRuFHASvJScdTaZjJdiSTqQXribjDYTjVS0T11fCLVrodbSgyGXR+OQkMxSIwAAoyADmrAgMlwGFkGFwNGFPDmkGDZm2whMVxgtwA0jBFAB3aB6AAy4T9GAAFvV6OqY0GQ8YeUJsAhk2mM9moHmC8X6gB2RqTIA)

Note: for .NET 7 the indices can be written in ascending order too.